spikeinterface/extractors/phykilosortextractors.py

```diff
@@ -109,17 +110,21 @@ def __init__(self, folder_path, exclude_cluster_groups=None, keep_good_only=Fals
             else:
                 max_si_unit_id = int(np.nanmax(unit_ids))
 
-            spike_clusters_new = np.zeros_like(spike_clusters_clean)
             for i, (phy_id, si_id) in enumerate(zip(cluster_info["cluster_id"].values,
-                                                    cluster_info["si_unit_id"].values)):
+                                                    cluster_info["si_unit_id"].values)):  # Slow
                 if np.isnan(si_id):
                     max_si_unit_id += 1
                     new_si_id = int(max_si_unit_id)
                 else:
                     new_si_id = si_id
                 unit_ids[i] = new_si_id
 
-                spike_clusters_new[spike_clusters_clean == phy_id] = new_si_id
+            # Little hack to replace values in spike_clusters_clean to spike_clusters_new very efficiently.
+            from_values = cluster_info['cluster_id'].values
+            sort_idx = np.argsort(from_values)
+            idx = np.searchsorted(from_values, spike_clusters_clean, sorter=sort_idx)
+            spike_clusters_new = unit_ids[sort_idx][idx]
```

Comment:
Hack found from https://stackoverflow.com/questions/19666626/replace-all-elements-of-python-numpy-array-that-are-greater-than-some-value